site/app/controllers/forum/ForumController.php

```diff
@@ -560,6 +564,51 @@ public function publishPost() {
                 $result['next_page'] = $this->core->buildCourseUrl(['forum', 'threads', $thread_id]) . '?' . http_build_query(['option' => $display_option]);
                 $result['post_id'] = $post_id;
                 $result['thread_id'] = $thread_id;
+                $posts = $this->core->getQueries()->getPostsInThreads([$result['thread_id']]);
+                $first = true;
+                $first_post_id = 1;
+                $order_array = [];
+                $reply_level_array = [];
+                if ($display_option !== 'tree') {
+                    $reply_level = 1;
+                }
+                else {
+                    foreach ($posts as $post) {
+                        if ($thread_id == -1) {
+                            $thread_id = $post["thread_id"];
+                        }
+                        if ($first) {
+                            $first = false;
+                            $first_post_id = $post["id"];
+                        }
+                        if ($post["parent_id"] > $first_post_id) {
+                            $place = array_search($post["parent_id"], $order_array, true);
+                            $tmp_array = [$post["id"]];
+                            $parent_reply_level = $reply_level_array[$place];
+                            if ($place !== false) {
+                                while (array_key_exists($place + 1, $reply_level_array) && $reply_level_array[$place + 1] > $parent_reply_level) {
+                                    $place++;
```

Comment:
While this data marshaling logic is ugly, I recall there being a reasonable reason for copying logic from elsewhere in the codebase when I discussed it with @ANS1514.  I'd like to see this get refactored eventually, but it doesn't need to block this PR if it functions correctly.